core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -103,13 +105,48 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   }
 
   /**
-   * Simplified version of combineByKey that hash-partitions the output RDD.
+   * This method is here for backward compatibility. It
+   * does not provide combiner classtag information to
+   * the shuffle.
```

Comment:
The java doc should start with a description of what the method does. We should use the old one and add that it exists backward compatibility after the first sentence.